src/Text/Pandoc/Readers/Markdown.hs

```diff
@@ -1735,6 +1736,17 @@ source = do
 linkTitle :: PandocMonad m => MarkdownParser m Text
 linkTitle = quotedTitle '"' <|> quotedTitle '\''
 
+-- Github wiki style link, with optional title
+githubWikiLink :: PandocMonad m => MarkdownParser m (F Inlines)
+githubWikiLink = try $ do
+  guardEnabled Ext_markdown_github_wikilink
+  string "[["
+  skipSpaces
+  let allUntil = manyTillChar (satisfy $ const True)
+  part1 <- allUntil (oneOf "|]")
+  part2 <- (char ']' >> return part1) <|> allUntil (string "]]")
+  returnF $ B.link part1 "" (B.text $ escapeURI $ trimr part2)
```

Comment:
In the MediaWiki reader, we insert a special title "wikilink" (if I recall) for wikilinks. This allows them to be distinguished in the AST, so people can do things like transform their URLs (adding a prefix for example).